### PR TITLE
Add handling for .DAS and .DDS calls

### LINKED ISSUE #42: handle DAS and DDS requests to RasterServer
Currently DAS and DDS requests made to `RasterServer` result in incorrect forwarding URLs. Update `build_orca_url()` to handle them, or perhaps make a separate function for generating forwarding URLs for DAS and DDS requests.

### PR DESCRIPTION
**Pull request by corviday**
This branch adds the ability to handle DDS and DAS metadata calls to the python3/THREDDS version of `pdp_util`, intended to support a python3 version of the PDP. 

Changes to make before merging:

- [x]  get rid of debug print statements
- [x] possibly add some tests of DDS and DAS functionality
- [ ] probably could rewrite request handling code more elegantly
- [x] black-ify

Resolves #42

### CODE REVIEW

File: pdp_util/raster.py [hunk 1]
```diff
@@ -86,12 +86,13 @@ def __call__(self, environ, start_response):
                 charset="utf-8",
             )
             return res(environ, start_response)
-
-        else:
-            orca_url = build_orca_url(
-                self.config["handlers"], self.config["thredds_root"], req
+        elif req.path_info.split(".")[-1] != "nc":
```
Comment (rod-glover) [inline pdp_util/raster.py:89]: May be simpler to use `if req.path_info.endswith(".nc")`

File: pdp_util/raster.py [hunk 2]
```diff
@@ -238,22 +239,80 @@ def __call__(self, environ, start_response):
...
-    [thredds_root]/[filepath]:[variable][time_start:time_end][lat_start:lat_end][lon_start:lon_end]
+    Orca is available through a url with one of the formats:
+    1. [orca_root]/?filepath=[filepath]
+    2. [orca_root]/?filepath=[filepath]&targets=time[time_start:time_end],lat[lat_start:lat_end],lon[lon_start:lon_end],[variable][time_start:time_end][lat_start:lat_end][lon_start:lon_end]
 
     where the [filepath] can be attained by the mapping of handler url to handler file from a config dict
     """
     filename = None
     for handler in handlers:
+        # print("handler url: {} path_info: {}".format(handler["url"], req.path_info[:-3]))
```
Comment (rod-glover) [inline pdp_util/raster.py:254]: Remove?

File: pdp_util/raster.py [hunk 3]
```diff
@@ -238,22 +239,80 @@ def __call__(self, environ, start_response):
...
+
+def get_target_dims(var):
+    """Adds the dimensions with the bounds matching the data variable to the targets
+    for orca data requests."""
+    try:
+        bounds = var[var.index("[") :]
+    except ValueError:  # Get full range of data variable
+        return "time,lat,lon,"
+
+    split_bounds = [bound + "]" for bound in bounds.split("]")][:-1]
```
Comment (rod-glover) [inline pdp_util/raster.py:277]: I'm having trouble figuring out exactly what is in `split_bounds` and what the ultimate value computed from it (`target_dims`) is. 

- Would this computation be simpler `bounds` were computed as `var[var.index("[") : -1]`? 
- Adding info in the docstring specifying the return value more exactly would help, plus maybe some intermediate comments in the body. 
- And/or tests of this function.

File: pdp_util/raster.py [hunk 4]
```diff
@@ -238,22 +239,80 @@ def __call__(self, environ, start_response):
...
 
-    Orca is available through a url with the format:
-    [thredds_root]/[filepath]:[variable][time_start:time_end][lat_start:lat_end][lon_start:lon_end]
+    Orca is available through a url with one of the formats:
+    1. [orca_root]/?filepath=[filepath]
+    2. [orca_root]/?filepath=[filepath]&targets=time[time_start:time_end],lat[lat_start:lat_end],lon[lon_start:lon_end],[variable][time_start:time_end][lat_start:lat_end][lon_start:lon_end]
 
     where the [filepath] can be attained by the mapping of handler url to handler file from a config dict
     """
     filename = None
```
Comment (rod-glover) [inline pdp_util/raster.py:252]: Should this inline computation be replaced by a call to `get_filepath_from_handlers` below?

File: tests/test_raster.py [hunk 5]
```diff
@@ -229,6 +229,98 @@ def test_RasterServer_orca(mm_database_dsn, config, environ, var):
...
+        Float64 time[time = 55115];
+        Float64 lat[lat = 510];
+        Float64 lon[lon = 1068];
+    } tasmin;
+} datasets/storage/data/climate/downscale/BCCAQ2/bccaqv2_with_metadata/tasmin_day_BCCAQv2%2bANUSPLIN300_inmcm4_historical%2brcp85_r1i1p1_19500101-21001231.nc;
+""",
+        ),
+    ],
+)
+@pytest.mark.online
```
Comment (rod-glover) [inline tests/test_raster.py:295]: Remove; this is a repeat of one at top.

File: pdp_util/raster.py [hunk 6]
```diff
@@ -86,12 +86,21 @@ def __call__(self, environ, start_response):
                 charset="utf-8",
             )
             return res(environ, start_response)
-
+        elif not req.path_info.split(".")[-1].endswith("nc"):
```
Comment (corviday) [inline pdp_util/raster.py:89]: I appreciate your documentation on the fancy new functions you added. Might be nice to also add a comment saying what the three branches of this if do, in high level terms. My guess is file list, .dds or .das metadata, and actual data?

File: pdp_util/raster.py [hunk 7]
```diff
@@ -238,25 +247,78 @@ def __call__(self, environ, start_response):
...
-    requests to THREDDS into bite-sized chunks and then reasemmbles them for the user.
+def build_orca_url(handlers, orca_root, thredds_root, req):
+    """orca is the OPeNDAP Request Compiler Application which pulls apart large OPeNDAP requests
+    to THREDDS into bite-sized chunks and then reasemmbles them for the user.
 
-    Orca is available through a url with the format:
-    [thredds_root]/?filepath=[filepath]&targets=[variable][time_start:time_end][lat_start:lat_end][lon_start:lon_end]
+    Orca is available through a url with one of the formats:
+    1. [orca_root]/?filepath=[filepath]&thredds_base=[thredds_root]
+    2. [orca_root]/?filepath=[filepath]&thredds_base=[thredds_root]&targets=time[time_start:time_end],lat[lat_start:lat_end],lon[lon_start:lon_end],[variable][time_start:time_end][lat_start:lat_end][lon_start:lon_end]
```
Comment (corviday) [inline pdp_util/raster.py:256]: :+1:

**Review by corviday — COMMENTED** [review]
Looks good to me. Code changes relatively straightforward and well documented.

Consider this an approval; I am unable to officially approve a pull request I opened (though I did not do most of the work!)

**Review by rod-glover — APPROVED** [review]
I have nothing new to add. I don't see any responses to my earlier review comments; perhaps they can be marked as done, ignored, etc. as the case may be.

Merge at will.

**Comment by eyvorchuk** [comment]
> I don't see any responses to my earlier review comments; perhaps they can be marked as done, ignored, etc. as the case may be.

Oh, sorry, I stealthily added various commits to address them. I implemented all your previous suggestions.

File: pdp_util/raster.py (hunk 6 above)
Comment (eyvorchuk) [reply pdp_util/raster.py:90]: Just added comments.

**Pull request merged**